### PR TITLE
Refactor node and npm middleware

### PR DESCRIPTION
**Pull request by anfibiacreativa**
- [x] Rename all instances of Node and Npm to JavaScript
- [x] Refactor middleware AddJavaScriptApp
- [x] Refactor middleware AddJavaScriptCLIApp
- [x] Refactor test
- [ ] Refactor other files

Opening as draft because I have not refactored some satellite files I'm not very sure the mechanism of
This is an attempt and must be run
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2341)

### CODE REVIEW

**Comment by pavelsavara** [comment]
cc @radical I was asked to help to redirect it

**Comment by mitchdenny** [comment]
What is the motivation for this change?

**Comment by spboyer** [comment]
@mitchdenny - see #1448

File: src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+
+using Microsoft.Extensions.Hosting;
```
Comment (mitchdenny) [inline src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs:6]: Remove this line.

File: src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="scriptPath">The path to the script that the runtime will execute.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<JavaScriptAppResource> AddJavaScriptApp(this IDistributedApplicationBuilder builder, string name, string? scriptPath = null, string? workingDirectory = null, string[]? args = null)
+    {   
+        scriptPath ??= "node";
```
Comment (mitchdenny) [inline src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs:26]: So if the value of `scriptPath` is not null ... lets say it is something like `myscript.js` - what is expected to happen here. Is the expectation that the JavaScript file itself will have a `#!/bin/...` prefix on it? How would that work on Windows?

File: .github/workflows/markdownlint.yml [hunk 3]
```diff
@@ -24,3 +24,5 @@ jobs:
         echo "::add-matcher::.github/workflows/markdownlint-problem-matcher.json"
         npm i -g markdownlint-cli
         markdownlint --ignore '.dotnet/' '**/*.md'
+
+# why is node.js version 16.x used?
```
Comment (mitchdenny) [inline .github/workflows/markdownlint.yml:28]: If this needs to be changed let's put it in a different PR because its orthogonal to what is being proposed here.

File: src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        return builder.AddResource(resource)
+                      .WithJavaScriptAppDefaults();
+    }
+
+    // this needs tested only for node since
+    // other runtimes may use a different pattern for env configuration
+     private static IResourceBuilder<JavaScriptAppResource> WithJavaScriptAppDefaults(this IResourceBuilder<JavaScriptAppResource> builder)
+    {
+        var environment = builder.ApplicationBuilder.Environment;
+        Console.WriteLine("Resource Command: " + builder.Resource.Command);
```
Comment (mitchdenny) [inline src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs:79]: I'm sure this is just temporary debug output but we don't do `Console.WriteLine(...)` in this methods. They generally don't have logging at all. These commands run before our DI container is spun up so we can't provide them a `ILogger<T>` either.

File: src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        /* if (builder.Resource.Command.Equals("node", StringComparison.OrdinalIgnoreCase))
```
Comment (mitchdenny) [inline src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs:80]: We would want to retain injecting OTLP environment variables (and presumably `NODE_ENV` too). We want to make sure as much as possible that code that is instrumented for OTEL just works.
